cmd/entrypoint/entrypoint.go

```diff
@@ -102,6 +102,10 @@ func Main(ctx context.Context, Version string, args ...string) error {
 		// Demo mode!
 		dlog.Infof(ctx, "DEMO MODE")
 		demoMode = true
+		// Set an environment variable so that other parts of the code can check if demo mode is active (mainly used for disabling synthetic authservice injection)
+		os.Setenv("AMBASSADOR_DEMO_MODE", "true")
```

Comment:
kind of nitpick/philosophical comment, I don't have too strong of opinions on this but it feels like we have env variable strings spread throughout the code base which makes it difficult to get a full list of what they are (short of hoping docs are up-to-date). 

This is a good example where at a minimum it might be good to bubble it up to the top of file so its visible to the package and can be used within the package.

`const envAmbassadorDemoMode string = "AMBASSADOR_DEMO_MODE"`.

Either prefix or postfix it, maybe as a team we can discuss preferences. The likely chance of it changing is low but this would make it easier in the future to change once (dry principle). The trade-off can be that you have to follow the variable to see the actual string...

Reply:
Sure, this makes sense and I think making it more visible by bubling it up to the top of the file (and prefixing it with env... will make things more searchable to look for all the environment variables). I'd say definitely worth the tradeoff of having to check the variable.